app/code/Magento/InventoryConfiguration/Model/GetStockItemConfiguration.php

```diff
@@ -73,12 +63,6 @@ public function __construct(
      */
     public function execute(string $sku, int $stockId)
     {
-        $stockItemData = $this->getStockItemData->execute($sku, $stockId);
-        if (null === $stockItemData) {
-            // Sku is not assigned to Stock
-            return null;
-        }
-
```

Comment:
looks like it would be great to throw `NoSuchEntityException` exception instead of removed code. It would be more reliable for a code of business logic.
But for doing this we need to introduce one more service `isProductAssignedToStock::execute($sku, $stockId)` to InventoryApi.

Because currently we create and return configuration with default values for a product which is actually not belonging to specified Stock